packages/plugins/other/visitor-plugin-common/src/selection-set-to-object.ts

```diff
@@ -102,7 +102,7 @@ export class SelectionSetToObject<Config extends ParsedDocumentsConfig = ParsedD
           this._appendToTypeMap(types, typeOnSchema.name, fields);
           this._appendToTypeMap(types, typeOnSchema.name, spreadsUsage[typeOnSchema.name]);
           this._collectInlineFragments(typeOnSchema, inlines, types);
-        } else if (isInterfaceType(typeOnSchema) && parentType.isTypeOf(typeOnSchema, null, null)) {
```

Comment:
I think this `isTypeOf` check is there to ensure that only fragments that can be applied to the `parentType` are added to the selection set. AFAIK according to the GraphQL spec it is possible to apply fragments on a type where it does not make sense, so this check should prevent generating incorrect types.

But it also seems like the isTypeOf method is actually optional. https://github.com/graphql/graphql-js/blob/711425eaf3baa3efa5da10e11ed76f58ec149ad7/src/type/definition.js#L729

I think the correct way would be to use the getInterfaces method and check whether the typeOnSchema is included in that array. https://github.com/graphql/graphql-js/blob/711425eaf3baa3efa5da10e11ed76f58ec149ad7/src/type/definition.js#L762

@benmosher could you try that?

Reply:
See as a reference: https://codesandbox.io/s/staging-brook-g1nrs?fontsize=14&hidenavigation=1&theme=dark

It spreads Avatar on the User type, however, execution does not complain 🤷